Library/Homebrew/rubocops/components_order.rb

```diff
@@ -79,14 +82,20 @@ def audit_formula(_node, _class_node, _parent_class_node, body_node)
             minimum_methods = allowed_methods.first.map { |m| "`#{m}`" }.to_sentence
             maximum_methods = allowed_methods.last.map { |m| "`#{m}`" }.to_sentence
 
-            on_os_bodies.each do |on_os_block, on_os_body|
-              method_name = on_os_block.method_name
-              child_nodes = on_os_body.begin_type? ? on_os_body.child_nodes : [on_os_body]
-              if child_nodes.all? { |n| n.send_type? || n.block_type? }
-                method_names = child_nodes.map(&:method_name)
-                next if allowed_methods.include? method_names
+            on_system_bodies.each do |on_system_block, on_system_body|
+              method_name = on_system_block.method_name
+              child_nodes = on_system_body.begin_type? ? on_system_body.child_nodes : [on_system_body]
+              if child_nodes.all? { |n| n.send_type? || n.block_type? || n.lvasgn_type? }
+                method_names = child_nodes.map do |node|
+                  next if node.lvasgn_type?
+                  next if node.method_name == :patch
+                  next if on_system_methods.include? node.method_name
```

Comment:
`on_{system}` blocks in `resource` blocks can now contain variable assignments, `patch` blocks, and other `on_{system}` blocks.